### ISSUE TITLE
DSM7: Repair of Radarr loses AppData configuration

### ISSUE THREAD

**Issue by mreid-tt** [SynoCommunity/spksrc#4759]
### Setup
_Package Name:_ Radarr
_Package Version:_ 3.2.2.5080

_NAS Model:_ DS916+
_NAS Architecture:_ INTEL Pentium
_DSM version:_ 7.0-41890

### Expected behavior
When you upgrade to DSM7 the apps need to be 'repaired' to function correctly. When I did the upgrade for Radarr it should preserve all the AppData as it did with Lidarr and Sonarr

### Actual behavior
It appears that while the AppData was moved from `/volume1/@appstore/radarr/var/.config/Radarr` to `/volume1/@appdata/radarr/.config/Radarr`, the Radarr configuration was still pointing to the old location

### Steps to reproduce
_1._ Upgrade DSM 6 to 7
_2._ Repair Radarr
_3._ Launch Radarr and check configuration

#### Package log
_Installer log: find logfile `/var/log/packages/{package}.log`, for DSM<6: `/var/packages/{package}/target/var/{package}_installer.log` and protocol in Package Center.
You find service log and other log files in `/var/packages/{package}/target/var`.
_
```
Tue Aug  3 22:09:39 -04 2021
Starting radarr command env HOME=/volume1/@appstore/radarr/var LD_LIBRARY_PATH=/volume1/@appstore/radarr/lib /volume1/@appstore/radarr/share/Radarr/bin/Radarr 
[Info] Bootstrap: Starting Radarr - /volume1/@appstore/radarr/share/Radarr/bin/Radarr - Version 3.2.2.5080 
[Warn] SingleInstancePolicy: Failed to check for multiple instances of Radarr. 

[v3.2.2.5080] System.UnauthorizedAccessException: Access to the path '/proc/28231/maps' is denied.
 ---> System.IO.IOException: Permission denied
   --- End of inner exception stack trace ---
   at System.IO.FileStream.CheckFileCall(Int64 result, Boolean ignoreNotSupported)
   at System.IO.FileStream.ReadNative(Span`1 buffer)
   at System.IO.FileStream.ReadSpan(Span`1 destination)
   at System.IO.FileStream.Read(Byte[] array, Int32 offset, Int32 count)
   at System.IO.StreamReader.ReadBuffer()
   at System.IO.StreamReader.ReadLine()
   at System.IO.ReadLinesIterator.MoveNext()
   at Interop.procfs.ParseMapsModulesCore(IEnumerable`1 lines)+MoveNext()
   at System.Diagnostics.ProcessManager.GetModules(Int32 processId)
   at System.Diagnostics.Process.get_Modules()
   at NzbDrone.Common.Processes.ProcessProvider.<>c__DisplayClass21_0.<GetProcessesByName>b__0(Process process) in D:\a\1\s\src\NzbDrone.Common\Processes\ProcessProvider.cs:line 347
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Set`1.UnionWith(IEnumerable`1 other)
   at System.Linq.Enumerable.UnionIterator`1.FillSet()
   at System.Linq.Enumerable.UnionIterator`1.ToList()
   at NzbDrone.Common.Processes.ProcessProvider.GetProcessesByName(String name)
   at NzbDrone.Common.Processes.ProcessProvider.FindProcessByName(String name) in D:\a\1\s\src\NzbDrone.Common\Processes\ProcessProvider.cs:line 94
   at Radarr.Host.SingleInstancePolicy.GetOtherNzbDroneProcessIds() in D:\a\1\s\src\NzbDrone.Host\SingleInstancePolicy.cs:line 68


[Info] Router: Application mode: Interactive 
[Info] MigrationController: *** Migrating data source=/volume1/@appstore/radarr/var/.config/Radarr/radarr.db;cache size=-20000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] MigrationController: *** Migrating data source=/volume1/@appstore/radarr/var/.config/Radarr/logs.db;cache size=-20000;datetimekind=Utc;journal mode=Wal;pooling=True;version=3 *** 
[Info] WebHostController: Listening on the following URLs: 
[Info] WebHostController:   http://*:8310 
[Info] RadarrBootstrapper: Starting Web Server 
[Info] CommandExecutor: Starting 2 threads for tasks. 
[Warn] DownloadMonitoringService: Unable to retrieve queue and history items from SABnzbd 

```
#### Other logs
_E.g. `/var/log/messages` or `/var/log/synopkg.log`_
```
N/A
```

**Comment by houndtt** [comment]
To resolve the missing configuration, I took the following steps:

1. Stop Radarr package
2. SSH into the NAS
3. Elevate to root with `sudo -i`
4. Backup new config with `mv /volume1/@appstore/radarr/var/.config/Radarr /volume1/@appstore/radarr/var/.config/Radarr.new`
5. Copy the previous config with `cp -rp /volume1/@appdata/radarr/.config/Radarr /volume1/@appstore/radarr/var/.config/Radarr`
6. Rename old config with `mv /volume1/@appdata/radarr/.config/Radarr /volume1/@appdata/radarr/.config/Radarr.old`
7. Exit from SSH session with `exit`
8. Start Radarr package

**Comment by publicarray** [comment]
might be fixed in #4757

**Comment by houndtt** [comment]
> might be fixed in #4757

i ran the new package in a VM as a manual upgrade. while the installation did change the AppData path correctly, the previous configuration from the already upgraded DSM7 install seemed to be lost. i don't know if this was a peculiarity with my setup but i will be taking a manual backup prior to upgrading when released just in case.

**Comment by hgy59** [comment]
Just added a section in README that `package repair` is known to fail and package settings are known to be lost.

**Comment by houndtt** [comment]
> Just added a section in README that `package repair` is known to fail and package settings are known to be lost.

I noted and thank you for your update. This is what I see on the page:
> ATTENTION: As reported, with a Package repair you will lose all the settings of installed packages. So make sure you backup the settings and configuration data before upgrading a package under DSM 7. A package repair will not succeed and you have to make a clean intallation on DSM 7 after completlty removing the former installation.

I would suggest different wording since my settings for Sonarr and Lidarr were fine following a repair. Additionally, a clean install was not necessary for me as the repair did not corrupt anything, I just needed to restore the configuration to the right location.

Perhaps it could be reworded like this:
> ATTENTION: As reported, package configuration settings may be lost following the upgrade to DSM 7 and the execution of a Package repair. Make sure to backup your settings and configuration for your SynoCommunity packages before installation of DSM 7 to facilitate restoration if needed

**Comment by hgy59** [comment]
@houndtt thanks, just updated the readme as you proposed.

**Comment by houndtt** [comment]
@publicarray, I decided to re-test the manual installation of #4757 that you mentioned. I was able to (on a clean DSM7 deployment) confirm that following a manual upgrade, the folder at `/volume1/@appstore/radarr/var/` is removed on deployment.

As such, the recommended way for users who may have already upgraded to DSM7 (prior to this new build being released) to retain their configurations would be:

1. Create manual backup via `System > Backup > Backup Now` and download the file (for safety)
2. Manually stop Radarr in the Package Center
3. Connect to DSM7 via SSH and run the following commands...
4. Archive any existing config using `sudo mv /volume1/@appdata/radarr/.config /volume1/@appdata/radarr/.config.old`
5. Move current configuration into place using `sudo mv /volume1/@appstore/radarr/var/.config /volume1/@appdata/radarr`
6. Exit the DSM7 via SSH using `exit`
7. Within Package Center install the new Radarr build (20210801-16 as of writing)
8. Launch Radarr and verify configurations are present (restore from manual backup if not)

Via the above method I was able to retain my previous configurations along with movie posters (which would normally be re-downloaded if using a manual restore from backup).

**Comment by houndtt** [comment]
One other thing I noted using a clean install of Virtual DSM7 (rather than the upgrade from DSM6) is that the in-app update for Radarr no longer gives the `/proc` folder access error for the Sonarr process if running as seen in the package log:

```
[v3.2.2.5080] System.UnauthorizedAccessException: Access to the path '/proc/28231/maps' is denied.
 ---> System.IO.IOException: Permission denied
```

This was an older issue I previously raised under https://github.com/Radarr/Radarr/issues/6068. I find this very strange since I can't seem to identify what is really different between the two `/proc` folders. When I check the clean install DSM7 `/proc` folder for the nzbdrone user process I see the following:

`dr-xr-xr-x 9 sc-nzbdrone sc-nzbdrone 0 Aug  8 07:00 10420`

When I check the existing upgrade from DSM6 `/proc` folder I see the following:

`dr-xr-xr-x 8 sc-nzbdrone nzbdrone 0 Aug  8 06:12 12926`

I really can't see much of a difference other than the group. Everything else looks the same. Who knows, maybe when my two remaining SynoCommunity apps (sabnzb and transmission) are upgradeable I may try backing them up and removing them all for a clean(ish) install to see if this changes the in-app update behaviour.

**Comment by iFredOS** [comment]
> @publicarray, I decided to re-test the manual installation of #4757 that you mentioned. I was able to (on a clean DSM7 deployment) confirm that following a manual upgrade, the folder at `/volume1/@appstore/radarr/var/` is removed on deployment.
> 
> As such, the recommended way for users who may have already upgraded to DSM7 (prior to this new build being released) to retain their configurations would be:
> 
> 1. Within Radarr, create manual backup via `System > Backup > Backup Now` and download the file (for safety)
> 2. Manually stop Radarr in the Package Center
> 3. Connect to DSM7 via SSH and run the following commands...
> 4. Archive any existing config using `sudo mv /volume1/@appdata/radarr/.config /volume1/@appdata/radarr/.config.old`
> 5. Move current configuration into place using `sudo mv /volume1/@appstore/radarr/var/.config /volume1/@appdata/radarr`
> 6. Exit the DSM7 via SSH using `exit`
> 7. Within Package Center install the new Radarr build ([20210801-16](tel:20210801-16) as of writing)
> 8. Launch Radarr and verify configurations are present (restore from manual backup if not)
> 
> Via the above method I was able to retain my previous configurations along with movie posters (which would normally be re-downloaded if using a manual restore from backup).

Hi. I tried to reinstall Radarr and the build “20210801-16” don’t exist, only “20210708-15”. Temporary I use the solution from @houndtt to be able to use my backup to run Radarr. Please let me know when this will be fixed in your packages server.

**Comment by mreid-tt** [comment]
This issue should be fixed with Radarr 20220515-18 as committed from https://github.com/SynoCommunity/spksrc/pull/5273. As such, this issue will be closed.
